third_party/freedom-typings/freedom-common.d.ts

```diff
@@ -118,7 +130,8 @@ declare module freedom {
     // We use this specification so that you can reference freedom sub-modules by
     // an array-lookup of it's name. One day, maybe we'll have a nicer way to do
     // this.
-    [moduleName:string] : Function;
+    // TODO: explore how to use FreedomModuleFactoryManager.
+    [moduleName:string] : any;
```

Comment:
What do you mean by "resolving this type"?  That sounds like it should work fine.